edweiss-app/components/lectures/remotecontrol/modal.tsx

```diff
@@ -60,33 +74,44 @@ const LangSelectModal: ReactComponent<{
         </ModalContainer>
     );
 };
-export default LangSelectModal;
-
 
 
+// ------------------------------------------------------------
+// ---------    Utils Sub Component For Lang Display    -------
+// ------------------------------------------------------------
 
 interface TwoLangsSelectionProps {
     lang: AvailableLangs;
     setLang: (lang: AvailableLangs) => void;
     lang1: AvailableLangs;
     lang2?: AvailableLangs;
 }
+/**
+ * Note: this component can handle an odd number of langs passed as input.
+ * You just would have to let lang2 tobe undefined
+ */
+const TwoLangsSelection: React.FC<TwoLangsSelectionProps & LightDarkProps> = ({ lang, setLang, lang1, lang2 }) => {
+    const theme = useTheme()
 
-const TwoLangsSelection: React.FC<TwoLangsSelectionProps> = ({ lang, setLang, lang1, lang2 }) => {
 
     /* Color pallette for selected/unselected languages
      * Note: this are `sky` and `green` in which there alpha has been modified
      */
-    const unselectedColorBord = 'rgba(4, 165, 229, 0.15)';
-    const unselectedColorBack = 'rgba(4, 165, 229, 0.01)';
-    const selectedColorBord = 'rgba(64, 160, 43, 0.6)';
-    const selectedColorBack = 'rgba(64, 160, 43, 0.1)';
+    const unselectedColorBord = (theme === "light") ? 'rgba(4, 165, 229, 0.15)' : 'rgba(166, 227, 161, 0.6)';
+    const unselectedColorBack = (theme === "light") ? 'rgba(4, 165, 229, 0.01)' : 'rgba(166, 227, 161, 0.15)';
+    const selectedColorBord = (theme === "light") ? 'rgba(64, 160, 43, 0.6)' : 'rgba(137, 220, 235, 0.7)';
+    const selectedColorBack = (theme === "light") ? 'rgba(64, 160, 43, 0.1)' : 'rgba(137, 220, 235, 0.3)';
```

Comment:
It's good as you have managing the interface when there is a light or dark theme. Maybe just a little suggestion that if you have a separate function to handle this (like a function in utilsFunctions), the code will look lots cleaner. If it is too much hassle then it is fine too to be honest.

Reply:
That makes sense! Keeping it close to where it's used does improve readability, especially for something as localized as this. I also get your point about not wanting to pollute the namespace unnecessarily. Maybe a middle ground could be grouping these kinds of contextual colors into a Colors section but with clear naming conventions that tie them to their specific use case?

But honestly, your approach works fine too if it's only used in one place—no point over-engineering it. 😊

Reply:
To be completly hones, I don't know how to do a `contextual namespace`, but yeah looking forward into it for sure. Thanks for the suggestion.